sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -335,6 +335,7 @@ def __init__(
         self._non_empty_ranges = None
         self._response = None
         self._encryption_data = None
+        self._offset = 0
```

Comment:
Does this `offset` property need to be checked/respected by any of the other download APIs?
I'm curious what would happen for example, if someone called `download.read(10)` followed by `download.readall()`.
While I consider this unlikely - if would be good to check what happens to a standard file IO, or bytes IO if you try to do this and compare our behaviour.

Reply:
As discussed offline, we did need to handle `offset` in other read APIs to align with standard IO. I've done that in the latest iteration.